### PR TITLE
Add test case if symlink works to variable tests

### PR DESCRIPTION
**Pull request by horike37**
## What did you implement:

Closes #4389 

The functionality introduced in #4389 lacks a unit test which checks working fine if you set up symlink.
I have added a test case, otherwise, we might be going to break that in future without noticing.

## How did you implement it:
Added a test which creating symlinks and read properly it via the variable system.

## How can we verify it:
Run `npm run test`

## Todos:

- [x] Write tests
- [ ] Write documentation
- [x] Fix linting errors
- [x] Make sure code coverage hasn't dropped
- [x] Provide verification config / commands / resources
- [x] Enable "Allow edits from maintainers" for this PR
- [x] Update the messages below

***Is this ready for review?:*** YES
***Is it a breaking change?:*** NO

### CODE REVIEW

File: lib/classes/Variables.test.js [hunk 1]
```diff
@@ -688,6 +688,23 @@ describe('Variables', () => {
...
     });
 
+    it('should populate symlinks', () => {
+      const serverless = new Serverless();
+      const SUtils = new Utils();
+      const tmpDirPath = testUtils.getTmpDirPath();
+      const realFilePath = path.join(tmpDirPath, 'someFile');
+      const symlinkPath = path.join(tmpDirPath, 'refSomeFile');
+
+      SUtils.writeFileSync(realFilePath, 'hello world');
```
Comment (HyperBrain) [inline lib/classes/Variables.test.js]: Shouldn't we remove the file and the symlink again after the test as a clean up step?

File: lib/classes/Variables.test.js [hunk 2]
```diff
@@ -688,6 +688,23 @@ describe('Variables', () => {
...
+      fse.ensureSymlinkSync(realFilePath, symlinkPath);
```
Comment (HyperBrain) [inline lib/classes/Variables.test.js]: We additionally should test if this works on Windows. I'm not sure what exactly `fse.ensureSymlinkSync` does there.

File: lib/classes/Variables.test.js [hunk 3]
```diff
@@ -688,6 +688,23 @@ describe('Variables', () => {
...
+
+      serverless.config.update({ servicePath: tmpDirPath });
+
+      return serverless.variables.getValueFromFile('file(./refSomeFile)').then(valueToPopulate => {
```
Comment (HyperBrain) [inline lib/classes/Variables.test.js]:
```
return expect(serverless.variables.getValueFromFile('file(./refSomeFile)')).to.be.fulfilled
.then(...
```

File: lib/classes/Variables.test.js (hunk 2 above)
Comment (horike37) [reply lib/classes/Variables.test.js]: Appveyor test has already passed, so this works fine on Windows as well.
https://ci.appveyor.com/project/serverless/serverless/build/1.0.682

**Comment by horike37** [comment]
Thaks @HyperBrain for the review :+1: 
I just updated the code as what you pointed out.
Please check again.

**Comment by horike37** [comment]
It looks like we've crisscrossed our action :smile: 
merging...

**Comment by HyperBrain** [comment]
:) Code is fine

**Pull request merged**